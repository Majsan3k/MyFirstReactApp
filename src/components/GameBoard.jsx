import React, {Component} from 'react';
import BoxComponent from "./BoxComponent";

export default class GameBoard extends Component{

    state = {
      availableMoves: true,
      boxClicked: false
    };

    render(){
        return (
            <div className="GameBoard">
                {this.props.components.map(box =>
                    <BoxComponent key={box.key} value={box.value}/>
                )}
            </div>
        )
    }
}