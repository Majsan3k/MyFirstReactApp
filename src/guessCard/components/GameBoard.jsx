import React, {Component} from 'react';
import Card from './Card';
import PropTypes from 'prop-types'
import {connect} from 'react-redux';
import {bindActionCreators} from "redux";
import {updateDeck, updateSecretCard} from "../../actions/guessCardActions";
import WinnerModal from "./WinnerModal";

class GameBoard extends Component {

    static propTypes = {
        cards: PropTypes.array.isRequired,
        secretCard: PropTypes.object.isRequired
    };

    state = {
        winner: false
    };

    componentWillMount() {
        this.updateDeck();
    }

    restart = () => {
        this.setState({winner: false}, this.updateDeck());
    };

    checkSecretCard = card => {
        if (card === this.props.secretCard.code) {
            this.setState({winner: true})
        }
    };

    updateDeck = () => {
        let index = Math.floor(Math.random() * 4);
        fetch('https://deckofcardsapi.com/api/deck/new/draw/?count=4')
            .then(res => res.json())
            .then(posts => this.props.updateDeck(posts))
            .then(posts => this.props.updateSecretCard(posts.payload.cards[index]));
    };

    render() {
        return (
            <div>
                {this.state.winner &&
                    <WinnerModal
                        history={this.props.history}
                        restart={this.restart}
                    />}
                <button onClick={this.updateDeck}>Shuffle</button>
                <div className='card-holder'>
                    {this.props.cards.map((card, index) =>
                        <Card src={card.images.png}
                              code={card.code}
                              key={index}
                              checkSecretCard={this.checkSecretCard}
                        />
                    )}
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    cards: state.guessCard.deck.cards,
    secretCard: state.guessCard.secretCard
});

const mapDispatchToProps = dispatch => ({
    updateDeck: bindActionCreators(updateDeck, dispatch),
    updateSecretCard: bindActionCreators(updateSecretCard, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(GameBoard)