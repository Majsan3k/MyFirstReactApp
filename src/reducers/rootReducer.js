import {combineReducers} from 'redux';
import twentyFortyEightReducer from './twentyFortyEightReducer';
import guessCardReducer from './guessCardReducer';

const rootReducer = combineReducers({
    twentyFortyEight: twentyFortyEightReducer,
    guessCard: guessCardReducer
});



export default rootReducer;