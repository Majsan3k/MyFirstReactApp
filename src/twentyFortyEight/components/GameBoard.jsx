import React, {Component} from 'react';
import BoxComponent from "./BoxComponent";
import PropTypes from 'prop-types'
import {connect} from 'react-redux'

class GameBoard extends Component{

    static propTypes = {
        components: PropTypes.array.isRequired
    };

    render(){
        return (
            <div className="GameBoard">
                {this.props.components.map((box) =>
                    <BoxComponent id={box.id}
                                  key={box.id}
                                  value={box.value}
                                  lastMove={this.props.lastMove}
                    />
                )}
            </div>
        )
    }
}

const mapStateToProps = state => ({
    components: state.twentyFortyEight.components,
    lastMove: state.twentyFortyEight.lastMove
});

export default connect(mapStateToProps, null)(GameBoard)