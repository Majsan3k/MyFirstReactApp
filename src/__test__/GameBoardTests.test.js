import React from 'react';
import GameBoard from '../twentyFortyEight/components/GameBoard';
import renderer from 'react-test-renderer';
import {createStore} from 'redux';
import rootReducer from '../reducers/rootReducer'
import {Provider} from 'react-redux';
import {shallow} from 'enzyme'

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('GameBoard', () => {

    let components = [];
    let store;
    for (let i = 0; i < 16; i++) {
        if (i === 4 || i === 7) {
            components.push({id: i, clicked: false, value: '2'})
        } else {
            components.push({id: i, clicked: false, value: ''})
        }
    }

    const initialState = {
        twentyFortyEight: {components: components, lastMove: ''}
    };

    beforeEach(() => {
        store = createStore(rootReducer, initialState);
    });

    it('should render correctly', () => {
        let container = renderer.create(<Provider store={store}><GameBoard components={components}/></Provider>)
            .toJSON();

        let wrapper = shallow(<Provider store={store}><GameBoard components={components}/></Provider>);
        expect(container).toMatchSnapshot();
    });
});