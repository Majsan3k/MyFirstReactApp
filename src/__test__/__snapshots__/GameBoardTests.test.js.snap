// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard should render correctly 1`] = `
<div
  className="GameBoard"
>
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
  <div
    className="box"
  >
    <p>
      2
    </p>
  </div>
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
  <div
    className="box"
  >
    <p>
      2
    </p>
  </div>
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
  <div
    className="box no-value"
  />
</div>
`;
